src/prover_disk.hpp

```diff
@@ -47,74 +47,52 @@ struct plot_header {
+        k(k),
+        table_begin_pointers(std::move(table_begin_pointers)),
+        C2(std::move(C2))
+    {
+        if (this->id.size() != kIdLen) {
+            throw std::invalid_argument("Invalid id size: " + std::to_string(this->id.size()));
         }
-        // 19 bytes  - "Proof of Space Plot" (utf-8)
-        // 32 bytes  - unique plot id
-        // 1 byte    - k
-        // 2 bytes   - format description length
-        // x bytes   - format description
-        // 2 bytes   - memo length
-        // x bytes   - memo
-
-        SafeRead(disk_file, (uint8_t*)&header, sizeof(header));
-        if (memcmp(header.magic, "Proof of Space Plot", sizeof(header.magic)) != 0)
-            throw std::invalid_argument("Invalid plot header magic");
-
-        uint16_t fmt_desc_len = Util::TwoBytesToInt(header.fmt_desc_len);
-
-        if (fmt_desc_len == kFormatDescription.size() &&
-            !memcmp(header.fmt_desc, kFormatDescription.c_str(), fmt_desc_len)) {
-            // OK
-        } else {
-            throw std::invalid_argument("Invalid plot file format");
+        if (k < kMinPlotSize || k > kMaxPlotSize) {
+            throw std::invalid_argument("Invalid k: " + std::to_string(k));
         }
-        memcpy(id.data(), header.id, sizeof(header.id));
-        this->k = header.k;
-        SafeSeek(disk_file, offsetof(struct plot_header, fmt_desc) + fmt_desc_len);
-
-        uint8_t size_buf[2];
-        SafeRead(disk_file, size_buf, 2);
-        memo.resize(Util::TwoBytesToInt(size_buf));
-        SafeRead(disk_file, memo.data(), memo.size());
-
-        this->table_begin_pointers = std::vector<uint64_t>(11, 0);
-        this->C2 = std::vector<uint64_t>();
-
-        uint8_t pointer_buf[8];
-        for (uint8_t i = 1; i < 11; i++) {
-            SafeRead(disk_file, pointer_buf, 8);
-            this->table_begin_pointers[i] = Util::EightBytesToInt(pointer_buf);
-        }
-
-        SafeSeek(disk_file, table_begin_pointers[9]);
-
-        uint8_t c2_size = (Util::ByteAlign(k) / 8);
-        uint32_t c2_entries = (table_begin_pointers[10] - table_begin_pointers[9]) / c2_size;
-        if (c2_entries == 0 || c2_entries == 1) {
-            throw std::invalid_argument("Invalid C2 table size");
+        if (this->table_begin_pointers.size() != nTableBeginPointerCount) {
+            throw std::invalid_argument("Invalid table_begin_pointers size: " + std::to_string(this->table_begin_pointers.size()));
         }
-
-        // The list of C2 entries is small enough to keep in memory. When proving, we can
-        // read from disk the C1 and C3 entries.
-        auto* c2_buf = new uint8_t[c2_size];
-        for (uint32_t i = 0; i < c2_entries - 1; i++) {
-            SafeRead(disk_file, c2_buf, c2_size);
-            this->C2.push_back(Bits(c2_buf, c2_size, c2_size * 8).Slice(0, k).GetValue());
+        uint32_t nExpectedSize = ((this->table_begin_pointers[10] - this->table_begin_pointers[9]) / (Util::ByteAlign(k) / 8)) - 1;
```

Comment:
this should be `const`

Reply:
See 94cad1af9eeebe6d2b5a0e80be3421b1aa6fd3ed